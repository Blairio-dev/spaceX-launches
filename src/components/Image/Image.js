import React from "react";
import PropTypes from "prop-types";

const Image = ({ alt, height, source, sourceSet, width }) => (
  <img
    alt={alt}
    height={height}
    src={source}
    srcSet={sourceSet}
    width={width}
  />
);

Image.propTypes = {
  alt: PropTypes.string.isRequired,
  height: PropTypes.string,
  source: PropTypes.string,
  sourceSet: PropTypes.string,
  width: PropTypes.string,
};

export { Image };
