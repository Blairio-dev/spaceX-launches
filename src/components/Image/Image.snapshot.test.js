import React from "react";
import renderer from "react-test-renderer";
import { Image } from "./Image";
import RefreshIcon from "../../assets/icon/refresh.png";
import RefreshIcon2x from "../../assets/icon/refresh@2x.png";
import RefreshIcon3x from "../../assets/icon/refresh@3x.png";

const DEFAULT_PROPS = {
  alt: "alternate-text",
  source: RefreshIcon,
};

const FULLEST_PROPS = {
  ...DEFAULT_PROPS,
  height: "64px",
  sourceSet: `
      ${RefreshIcon3x} 3x,
      ${RefreshIcon2x} 2x,
      ${RefreshIcon} 1x,
    `,
  width: "64px",
};

describe("Image Snapshots", () => {
  it("Renders with default props", () => {
    const tree = renderer.create(<Image {...DEFAULT_PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders with fullest props", () => {
    const tree = renderer.create(<Image {...FULLEST_PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
