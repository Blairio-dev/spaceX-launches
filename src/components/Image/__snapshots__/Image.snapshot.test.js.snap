// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Snapshots Renders with default props 1`] = `
<img
  alt="alternate-text"
  src="refresh.png"
/>
`;

exports[`Image Snapshots Renders with fullest props 1`] = `
<img
  alt="alternate-text"
  height="64px"
  src="refresh.png"
  srcSet="
      refresh@3x.png 3x,
      refresh@2x.png 2x,
      refresh.png 1x,
    "
  width="64px"
/>
`;
